### PR TITLE
Button 

### PR DESCRIPTION
**Pull request by leo-he-1**

### CODE REVIEW

**Review by Nap1400 — APPROVED** [review]

File: mobile/src/components/button/Button.view.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  let buttonClickableStyle;
+  isDisabled
+    ? (buttonClickableStyle = styles.buttonDisabled)
+    : (buttonClickableStyle = styles.buttonEnabled);
+
+  return (
+    <TouchableOpacity
+      style={[styles.container, buttonStyle, buttonClickableStyle]}
+      disabled={isDisabled}
+      onPress={() => press?.()}>
```
Comment (raynesmike) [inline mobile/src/components/button/Button.view.tsx:22]: Im assuming this will be a TODO once for the event catch in the future?

**Review by raynesmike — APPROVED** [review]
I approve this addition

**Comment by carizzuti** [comment]
Can you add a screenshot so we can see what it looks like?

**Comment by leo-he-1** [comment]
image

File: mobile/package.json [hunk 2]
```diff
@@ -11,7 +11,8 @@
   },
   "dependencies": {
     "react": "17.0.2",
-    "react-native": "0.68.1"
+    "react-native": "0.68.1",
+    "yarn": "^1.22.18"
```
Comment (lasharela) [inline mobile/package.json:15]: I think yarn needs to be installed globally . never seen yarn as dependence .. Maximum you can include as dev-dependency

File: mobile/src/components/button/Button.style.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import {StyleSheet} from 'react-native';
+
+export const styles = StyleSheet.create({
+  container: {
+    borderRadius: 28,
+    alignItems: 'center',
+    justifyContent: 'center',
+    width: 258,
```
Comment (lasharela) [inline mobile/src/components/button/Button.style.ts:8]: we are not setting fix widths... remember? what if mobile will be smaller?

File: mobile/src/components/button/Button.style.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    height: 60,
+  
+  },
+  buttonDisabled: {
+    backgroundColor: 'blue',
```
Comment (lasharela) [inline mobile/src/components/button/Button.style.ts:13]: let's create theme/color.ts and `theme/size.ts` as we had before. why not?

File: mobile/src/components/button/Button.view.tsx [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+  title  = 'Button',
+  isDisabled = false,
+  press, 
+  buttonStyle,
+  titleStyle,
+}) => {
+  let buttonClickableStyle;
+  isDisabled
+    ? (buttonClickableStyle = styles.buttonDisabled)
+    : (buttonClickableStyle = styles.buttonEnabled);
```
Comment (lasharela) [inline mobile/src/components/button/Button.view.tsx:16]: let buttonClickableStyle;
  isDisabled
    ? (buttonClickableStyle = styles.buttonDisabled)
    : (buttonClickableStyle = styles.buttonEnabled);
    
    why not:
    ```
    let buttonClickableStyle =  isDisabled ?  styles.buttonDisabled :  styles.buttonEnabled
    ```
    
    One line. clean... why you are doing longer?

**Review by lasharela — APPROVED** [review]

File: mobile/src/components/button/Button.type.ts [hunk 6]
```diff
@@ -0,0 +1,13 @@
+import  {StyleProp, ViewStyle} from 'react-native';
+
+export type ButtonProps = {
```
Comment (lasharela) [inline mobile/src/components/button/Button.type.ts:3]: `TButtonProps`
`IButton`

File: mobile/src/components/button/Button.view.tsx (hunk 1 above)
Comment (lasharela) [inline mobile/src/components/button/Button.view.tsx:22]: onPress={() => press()}>....but

onPress={press}>

Comment (lasharela) [reply mobile/src/components/button/Button.view.tsx:22]: ()=>press()

onPress: ()=>void   `()=>void `

**Review by lasharela — CHANGES_REQUESTED** [review]

**Pull request closed without merging**